tests/test_ace.rb

```diff
@@ -18,251 +18,211 @@
-    all_aces[afi][acl_name].each do |seqno, _inst|
-      next unless seqno.to_i == @seqno.to_i
-      found = true
+  def test_http_method
+    afi = 'ipv4'
+    %w(connect delete get head post put trace).each do |val|
+      a = ace_helper(afi, http_method: val)
+      assert_equal(val, a.http_method)
     end
+  end
 
-    @default_show_command =
-      "show runn aclmgr | sec '#{afi_cli} access-list #{acl_name}'"
-    assert(found,
-           "#{afi_cli} acl #{acl_name} seqno #{@seqno} is not configured")
+  def test_log
+    %w(ipv4 ipv6).each do |afi|
+      a = ace_helper(afi, log: true)
+      assert(a.log)
+      a = ace_helper(afi, log: false)
+      refute(a.log)
+    end
+  end
 
-    if entry.include?(:action)
-      action = "#{entry[:action]} .*"
-    else
-      action = "remark #{entry[:remark]}"
+  def test_precedence
+    afi = 'ipv4'
+    %w(critical flash flash-override immediate internet network
+       priority routine).each do |val|
+      a = ace_helper(afi, precedence: val)
+      assert_equal(val, a.precedence)
     end
-    assert_show_match(pattern: /\s+#{@seqno} #{action}$/,
-                      msg:     "failed to create ace seqno #{@seqno}")
-    ace.destroy
-    refute_show_match(pattern: /\s+#{@seqno} #{entry[:action]} .*$/,
-                      msg:     "failed to remove ace seqno #{@seqno}")
   end
 
-  def test_ace_update
-    action = 'permit'
-    proto = 'tcp'
-    src = '1.0.0.0/8'
-    dst = '3.0.0.0 0.0.0.8'
-    entry = { action: action, proto: proto, src_addr: src, dst_addr: dst }
+  def test_redirect
+    afi = 'ipv4'
+    val = 'port-channel1,port-channel2'
+    a = ace_helper(afi, redirect: val)
+    assert_equal(val, a.redirect)
+  end
 
-    a = Ace.new('ipv4', 'ace_update', 10)
-    a.ace_set(entry)
+  def test_tcp_option_length
+    afi = 'ipv4'
+    %w(0 16 28).each do |val|
+      a = ace_helper(afi, tcp_option_length: val)
+      assert_equal(val, a.tcp_option_length)
+    end
+  end
 
-    assert_equal(src, a.src_addr)
-    assert_equal(dst, a.dst_addr)
+  def test_time_range
+    val = 'my_range'
+    %w(ipv4 ipv6).each do |afi|
+      a = ace_helper(afi, time_range: val)
+      assert_equal(val, a.time_range)
+    end
+  end
 
-    src = '2.0.0.0/16'
-    entry = { action: action, proto: proto, src_addr: src, dst_addr: dst }
-    a.ace_set(entry)
-    assert_equal(src, a.src_addr)
+  def test_packet_length
+    val = 'range 80 1000'
+    %w(ipv4 ipv6).each do |afi|
+      a = ace_helper(afi, packet_length: val)
+      assert_equal(val, a.packet_length)
+    end
+  end
 
-    dst = '3.0.0.0 0.0.0.4'
-    entry = { action: action, proto: proto, src_addr: src, dst_addr: dst }
-    a.ace_set(entry)
-    assert_equal(dst, a.dst_addr)
+  def test_ttl
+    val = '3'
+    %w(ipv4 ipv6).each do |afi|
+      a = ace_helper(afi, ttl: val)
+      skip("This property has a known defect on the platform itself -\n"\
+           'CSCuy47463: access-list ttl does not nvgen') if a.ttl.nil?
```

Comment:
I'm fine with this for now, but in general we don't want to include DDTS numbers in our code since we have to scrub them out before the release and we might forget.